worlds/sc2/item/item_tables.py

```diff
@@ -2016,12 +2016,22 @@ def get_full_item_list():
     for number in range(len(SC2Mission))
 }
 # Flavor keys (key offset + 3000 - 3999)
+# Generic progressive keys are also in here because flavor keys don't dynamically scale with non-item data anyway
+# and just reserve a range for convenience
 flavor_key_item_table = {
+    item_names.PROGRESSIVE_MISSION_KEY:
+        ItemData(0 + SC2_KEY_ITEM_ID_OFFSET + SC2_KEY_ITEM_SECTION_SIZE * 3, FactionlessItemType.Keys, 0, SC2Race.ANY,
+                 classification=ItemClassification.progression, quantity=0),
+    item_names.PROGRESSIVE_QUESTLINE_KEY:
+        ItemData(1 + SC2_KEY_ITEM_ID_OFFSET + SC2_KEY_ITEM_SECTION_SIZE * 3, FactionlessItemType.Keys, 0, SC2Race.ANY,
+                 classification=ItemClassification.progression, quantity=0),
+}
+flavor_key_item_table.update({
     item_names._TEMPLATE_FLAVOR_KEY.format(name):
-        ItemData(i + SC2_KEY_ITEM_ID_OFFSET + SC2_KEY_ITEM_SECTION_SIZE * 3, FactionlessItemType.Keys, 0, SC2Race.ANY,
+        ItemData(i + 2 + SC2_KEY_ITEM_ID_OFFSET + SC2_KEY_ITEM_SECTION_SIZE * 3, FactionlessItemType.Keys, 0, SC2Race.ANY,
                  classification=ItemClassification.progression, quantity=0)
     for (i, name) in enumerate(item_names._flavor_key_names)
-}
+})
```

Comment:
Instead of shifting existing keys (and just reserving 2 spots for those) I'd just move the new ones into a new range so future ones can be easily added too. Just move the added ones to 7000+